### DISCUSSION TITLE
homebrew ruby: gem command not found

### DISCUSSION THREAD

**Discussion by rnkhouse** [rbenv/rbenv#1536] [Q&A]
Installing gem using rbenv in a dockerfile.

These are the steps:

    RUN brew install rbenv ruby-build
    RUN echo 'eval "$(rbenv init - bash)"' >> ~/.bashrc
    RUN rbenv install 3.2.2
    RUN rbenv global 3.2.2
    ENV PATH="$HOME/.rbenv/versions/3.2.2/bin:$HOME/.rbenv/shims:$PATH"

But, when I try to run gem command it's giving **error: gem command not found**.

    RUN gem install bundler

**Comment by rnkhouse** [comment]
Why the gem command is not working? Can anyone help, please?

**Comment by mislav** [comment]
I'm not a Docker expert (I barely ever use it), but I'm pretty sure that the command invoked by `RUN` doesn't run in bash shell, and definitely doesn't source an initialization file like `.bashrc`. You can remove the line that edits that file, since you already set up rbenv in PATH manually.

```
# DELETE THIS
RUN echo 'eval "$(rbenv init - bash)"' >> ~/.bashrc
```

Also, the PATH line should *not* have the `$HOME/.rbenv/versions/3.2.2/bin` directory, and instead should be:
```
ENV PATH="$HOME/.rbenv/shims:$PATH"
```

See https://github.com/rbenv/rbenv/wiki/Deploying-with-rbenv for more information. Note that if you're only going to have 1 Ruby version inside your Docker container, it makes little sense to install a version manager like rbenv. You might have an easier time just basing your Docker imagine on one of the available Ruby images:
```
FROM ruby:3.2
```

But, I'm not entirely sure why `gem` wasn't found. The `rbenv install` command was supposed to generate a file at `$HOME/.rbenv/shims/gem` that should have been in PATH already. You can use a command like this to debug things:

```
RUN echo $PATH && ls $HOME/.rbenv/shims/gem
```

Let us know if you find out!
